pkg/endpoint/endpoint.go

```diff
@@ -1310,10 +1314,49 @@ func (e *Endpoint) SetContainerName(name string) {
 	e.Mutex.Unlock()
 }
 
-// SetPodName modifies the endpoint's pod name
-func (e *Endpoint) SetPodName(name string) {
+// GetK8sNamespace returns the name of the pod if the endpoint represents a
+// Kubernetes pod
+func (e *Endpoint) GetK8sNamespace() string {
+	e.Mutex.RLock()
+	defer e.Mutex.RUnlock()
+
+	return e.k8sNamespace
+}
+
+// SetK8sNamespace modifies the endpoint's pod name
+func (e *Endpoint) SetK8sNamespace(name string) {
+	e.Mutex.Lock()
+	e.k8sNamespace = name
+	e.Mutex.Unlock()
+}
+
+// GetK8sPodName returns the name of the pod if the endpoint represents a
+// Kubernetes pod
+func (e *Endpoint) GetK8sPodName() string {
+	e.Mutex.RLock()
+	defer e.Mutex.RUnlock()
+
+	return e.k8sPodName
+}
+
+// GetK8sNamespaceAndPodName returns the namespace and pod name
+func (e *Endpoint) GetK8sNamespaceAndPodName() string {
```

Comment:
I see no usage of that method in the whole PR, `GetK8sNamespaceAndPodNameLocked` is used everywhere instead. Is it a secure method with mutex which is going to be used in the future?

Reply:
Good catch, this is a leftover and all uses have been replaced with `GetK8sNamespaceAndPodNameLocked`. Will remove.

Reply:
Oh, if we're only left with the *Locked function, perhaps make it unexported?

Reply:
actually, no, it's used by the endpoint manager... Should that use the locked version?

Reply:
Yes, it holds the endpoint lock. This is another example where we have an external package access Endpoint fields directly. We should move away from this long term.

Reply:
This one is a good example of a tough one, though. These various maps need to be kept in sync but they live with the manager. This is a good next-use-case for controllers, though: Introduce a deleted flag to the endpoint and have controllers for each of these lists clean up the ones that are deleted async (or even once the endpoint is done with cleanup)